src/libraries/Common/src/System/Security/Cryptography/KeyFormatHelper.cs

```diff
@@ -193,7 +194,14 @@ internal static AsnWriter WritePkcs8(
             algorithmIdentifierWriter.CopyTo(writer);
 
             // PKI.privateKey
-            using (writer.PushOctetString())
+            if (wrapPrivateKeyInOctetString)
```

Comment:
I don't quite understand this. In PKCS#8 RFC 5208, the ASN.1 syntax is

```

PrivateKeyInfo ::= SEQUENCE {
        version                   Version,
        privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
        privateKey                PrivateKey,
        attributes           [0]  IMPLICIT Attributes OPTIONAL }

      Version ::= INTEGER

      PrivateKeyAlgorithmIdentifier ::= AlgorithmIdentifier

      PrivateKey ::= OCTET STRING

      Attributes ::= SET OF Attribute
```

The `PrivateKey` is always an OCTET STRING. Why does it need to be optional?

Reply:
Okay, I get it. The SLH-DSA private key is just unstructured bytes, which you can't put in the incoming writer.

Reply:
Yes, that was the initial reason but I ended up removing the call to this helper. It made the code more complicated in the SLH-DSA case so I just inlined this logic. I'll revert this change.